### ISSUE TITLE
Tenant Sharding Support

### ISSUE THREAD

**Issue by Ikki-Dai** [apache/shardingsphere#14316]
### Is your feature request related to a problem?
NO
### Describe the feature you would like.
1. Not need database support Teanat feature 
2. Dynantic add or remove tenant's datasource and load sharding rules for each tenant
3. Execute tenant sharding before other sharding rules.

**Comment by tuichenchuxin** [comment]
Maybe you can try sql hint
https://shardingsphere.apache.org/document/current/cn/user-manual/shardingsphere-jdbc/special-api/sharding/hint/

**Comment by Ikki-Dai** [comment]
yes, but prefer concept of unity about teanat and something work around

**Comment by tuichenchuxin** [comment]
@Ikki-Dai I got you point. We'll have a think of it.

**Comment by terrymanu** [comment]
This issue is opened for long time any no feedback, I just close it.

**Comment by Ikki-Dai** [comment]
hibernate 6.0 have a @TenantId annotation to support something around tenant
